### PR TITLE
Update from upstream (after v0.40.0)

### PR DESCRIPTION
**Pull request by evan-forbes**
This PR simply updates to the latest upstream master branch from cosmos/cosmos-sdk (commit: https://github.com/cosmos/cosmos-sdk/commit/4fa9e98095a6d5fc6a045adfe225f498854206bb)

closes: #XXXX

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [X] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.

This PR is strictly maintenance, and while being required for at least Issues #1 and #2, it relates to most currently open issues

- [X] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [X] Wrote unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [X] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [X] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [X] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [X] Re-reviewed `Files changed` in the Github PR explorer
- [X] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

File: docs/run-node/interact-node.md [hunk 1]
```diff
@@ -52,12 +52,13 @@ You should see two delegations, the first one made from the `gentx`, and the sec
...
+Since the code generation library largely depends on your own tech stack, we will only present three alternatives:
 
 - `grpcurl` for generic debugging and testing,
+- programmatically via Go,
 - CosmJS for JavaScript/TypeScript developers.
 
-### grpcurl: Reflection, Queries, and Simulation
+### grpcurl
 
 [grpcurl])https://github.com/fullstorydev/grpcurl is like `curl` but for gRPC. It is also available as a Go library, but we will use it only as a CLI command for debugging and testing purposes. Follow the instructions in the previous link to install it.
```
Comment (liamsi) [inline docs/run-node/interact-node.md:63]: No need to apply this here but we could prob. open a PR with doc improvements / minor fixes to the SDK:
```suggestion
[grpcurl](https://github.com/fullstorydev/grpcurl) is like `curl` but for gRPC. It is also available as a Go library, but we will use it only as a CLI command for debugging and testing purposes. Follow the instructions in the previous link to install it.
```

File: go.sum [hunk 2]
```diff
@@ -562,6 +562,8 @@ github.com/tendermint/tendermint v0.34.0-rc4/go.mod h1:yotsojf2C1QBOw4dZrTcxbyxm
 github.com/tendermint/tendermint v0.34.0-rc6/go.mod h1:ugzyZO5foutZImv0Iyx/gOFCX6mjJTgbLHTwi17VDVg=
 github.com/tendermint/tendermint v0.34.0 h1:eXCfMgoqVSzrjzOj6clI9GAejcHH0LvOlRjpCmMJksU=
 github.com/tendermint/tendermint v0.34.0/go.mod h1:Aj3PIipBFSNO21r+Lq3TtzQ+uKESxkbA3yo/INM4QwQ=
+github.com/tendermint/tendermint v0.34.1 h1:TsJXY8+1NThc/XIENMv1ENzazriHBJJjeP6hN3kpC8g=
+github.com/tendermint/tendermint v0.34.1/go.mod h1:Aj3PIipBFSNO21r+Lq3TtzQ+uKESxkbA3yo/INM4QwQ=
```
Comment (liamsi) [inline go.sum:566]: Can you double-check if running `go mod tidy` changes anything in `go.sum`?

**Review by liamsi — APPROVED** [review]
Dope! Thanks Evan 👍🏼 LGTM

File: go.sum (hunk 2 above)
Comment (evan-forbes) [reply go.sum:566]: double checked, and there is no change

**Pull request merged**